flink-connector-mysql-cdc/src/main/java/com/ververica/cdc/connectors/mysql/source/utils/RecordUtils.java

```diff
@@ -328,7 +328,7 @@ public static TableId getTableId(SourceRecord dataRecord) {
         // the split key field contains single field now
         String splitFieldName = nameAdjuster.adjust(splitBoundaryType.getFieldNames().get(0));
         Struct key = (Struct) dataRecord.key();
-        return new Object[] {key.get(splitFieldName)};
+        return new Object[] {key.get(key.schema().fields().get(0))};
```

Comment:
Which case will lead this error @mincwang ?

Reply:
please look the #573